tests/unit_test/linux/README.md

```diff
@@ -0,0 +1,142 @@
+```cmake
+if (AFR_ENABLE_UNIT_TESTS)
+    add_subdirectory(folder_name/project_name)
+    return()
+endif()
+```
+3. In the directory of the module to be tested create a new directory called *utest*
+
+4. For simplicity`copy the CMakeLists.txt file from *libraries/abstractions/secure_sockets/utest/* into the new Unit Test Directory *utest*
+
+5. Create a test soure file that ends with  *_utest.c* it is where your test code will live<br>
+    you will end up with 2 files in the utest directory (project_name_utest.c and CMakeLists.txt)
+
+6. Edit the copied file (CMakeLists.txt) in the following way
+    1. Modify the *project_name* at the top to match the new one to be tested
+    2. Replace all headers in *mock_list* with the ones you the source file
+       under test uses.
+       ``` cmake
+        list(APPEND mock_list
+                    my_file.h
+                )
+       ```
+    3. Replace all the include directories in *mock_include_list* with the
+       directories your mocks are using
+       ```cmake
+        list(APPEND mock_include_list
+                    /my/file/dir/needs
+                )
+        ```
+    4. Replace the definitions in *mock_define_list* list with the definitions you see fit for your
+       source files
+       ``` cmake
+       list(APPEND mock_define_list
+                    -DMY_DEFINE=1
+                )
+       ```
+       This will create header files in the following form:
+       my_file_mock.h which you will have to include from
+       your test code (project_name_utest.c)
+       This compilation step would create a shared object with the name *project_name_mock.so*
+
+    5. Replace all the files in *real_source_files* with the source files you will
+       be testing. This will create a library with the name *project_file_real.a*
+       ``` cmake
+       list(APPEND real_source_files
+                    code_to_test.c
+                )
+       ```
+    6. Replace all the dirctories in *real_include_directories*  with the
+       directories the source under test will include headers from
+       ```cmake
+       list(APPEND real_include_directories
+                   /my/under/test/source/needs
+               )
+       ```
+       This compilation step would create an object library called *project_name_real.a*
+    7. Replace all directories in *test_include_directories* with what the test
+       code needs (project_name_utest.c)
+       ```cmake
+           list(APPEND test_include_directories
+                    /my/test/code/needs
+                )
+        ```
+
+    8. And thats it for this file..
+
+    9. At the end, this makefile will create the mock files and 3 object files
+       *project_name_mock.so*, *project_name_real.a* and *project_name_utest*
+
+7. Write your Unit Test code *project_name_utest.c* it is better if you copy a
+   file from another place *secure_sockets* for example to reuse its skeleton <br>
```

Comment:
(Nit) "...file from another place (for example, *secure_sockets*) to re-use its skeleton"